### ISSUE TITLE
Wrong indent warnings for neovim

### ISSUE THREAD

**Issue by rockyzhang24** [CppCXY/EmmyLuaCodeStyle#32]
I already posted the issue at [lua-language-server](https://github.com/sumneko/lua-language-server) repo (please see [here](https://github.com/sumneko/lua-language-server/issues/1003)), but I am not sure whether I should bring it here as well.

**Describe the bug**

I am using the formatter and code style check. Here is my config. 

```lua
nvim_lsp.sumneko_lua.setup {
  cmd = { "lua-language-server", "--preview" },
  settings = {
    Lua = {
      runtime = {
        -- Tell the language server which version of Lua you're using (most likely LuaJIT in the case of Neovim)
        version = 'LuaJIT',
        -- Setup your lua path
        path = runtime_path,
      },

      diagnostics = {

        neededFileStatus = {
          ["codestyle-check"] = "Any",
        },

      },

      format = {
        enable = true,
        defaultConfig = {
          indent_style = 'space',
          indent_size = 2,
        },
      },

      workspace = {
        -- Make the server aware of Neovim runtime files
        library = vim.api.nvim_get_runtime_file("", true),
      },
      -- Do not send telemetry data containing a randomized but unique identifier
      telemetry = {
        enable = false,
      },
    },
  },
}
```

As the config above, my indentation size is 2 spaces. However, it gives me wrong warnings for the indentations in functions.

<img width="1083" alt="image" src="https://user-images.githubusercontent.com/11582667/160215561-7558ca43-3578-44a1-b987-388afb9c998a.png">

**Environment (please complete the following information):**
 - OS: macOS (the latest version)
 - Client: neovim

Thank you so much.

**Comment by CppCXY** [comment]
In VSCode, it work well:
![image](https://user-images.githubusercontent.com/40318218/160224050-fe7ea5a8-9fca-433d-92d7-475037a312e3.png)
![image](https://user-images.githubusercontent.com/40318218/160224056-278a20b8-cb31-4a8a-a326-bbe7d791bded.png)

in neovim may be configured incorrectly, resulting in the default options not changing at all.

formatting works correctly because sumneko requires  using lsp-passed parameters for indentation of the current file.

maybe you can create an  .editorconfig file to fix this

**Comment by rockyzhang24** [comment]
Thanks. Yes, .editorconfig works well, however setting it in a lsp setup() function doesn't. I need the default config for every project, so .editorconfig file is not ideal for me. 

> in neovim may be configured incorrectly, resulting in the default options not changing at all

I think my config I attached above is correct, because only `indent_size` option doesn't work. If I set other options like `quote_style` as below, it works well. 

```lua
nvim_lsp.sumneko_lua.setup {
  cmd = { "lua-language-server", "--preview" },
  settings = {
    Lua = {
      runtime = {
        -- Tell the language server which version of Lua you're using (most likely LuaJIT in the case of Neovim)
        version = 'LuaJIT',
        -- Setup your lua path
        path = runtime_path,
      },

      diagnostics = {

        neededFileStatus = {
          ["codestyle-check"] = "Any",
        },

      },

      format = {
        enable = true,
        defaultConfig = {
          quote_style = "double",
        },
      },

      workspace = {
        -- Make the server aware of Neovim runtime files
        library = vim.api.nvim_get_runtime_file("", true),
      },
      -- Do not send telemetry data containing a randomized but unique identifier
      telemetry = {
        enable = false,
      },
    },
  },
}
```

The weirder thing is that, please see my screenshot above, the warnings are only shown for the indentation in function definition, not other places. 

Emm, so weird.

**Comment by rockyzhang24** [comment]
Wow, I figured it out. From [here](https://github.com/sumneko/lua-language-server/blob/d57c21e585e7285583b7379e9e955da3b1345508/script/config/config.lua#L212), the value in `defaultConfig` table should be `string`, not `integer`. So to make it work, we should use `indent_size = "2"` instead of `indent_size = 2`. 

Thanks anyway. I will close this.
